src/collective/mustread/db.py

```diff
@@ -29,12 +29,26 @@ def getEngine(conn_string=None, req=None):
             audit_conn_string = registry['collective.auditlog.interfaces.IAuditLogSettings.connectionstring']  # noqa
         except KeyError:
             audit_conn_string = None
-        if conn_string == audit_conn_string \
-           and req and 'sa.engine' in req.environ:
-                # re-use collective.auditlog connector if possible
-                engine = req.environ['sa.engine']
+        if conn_parameters is None:
+            conn_parameters = registry['collective.mustread.interfaces.IMustReadSettings.connectionparameters']  # noqa
+        try:
+            audit_conn_parameters = registry['collective.auditlog.interfaces.IAuditLogSettings.connectionparameters']  # noqa
+        except KeyError:
+            audit_conn_parameters = None
+
+        if (
+            conn_string == audit_conn_string
+            and conn_parameters == audit_conn_parameters
+            and req and 'sa.engine' in req.environ
+        ):
+            # re-use collective.auditlog connector if possible
+            engine = req.environ['sa.engine']
         else:
-            engine = create_engine(conn_string, echo=DEBUG)
+            if not conn_parameters:
+                conn_parameters = {}
+            elif isinstance(conn_parameters, basestring):
+                conn_parameters = loads(conn_parameters)
+            engine = create_engine(conn_string, **conn_parameters)
```

Comment:
The DEBUG convenience is lost on purpose... If you want to echo you can tweak the registry.